### PR TITLE
Use a typedef for the block index, and reduce it from 64 to 32 bit wh…

### PR DESCRIPTION
**Pull request by baldersheim**
…ich is plenty. Currently only 19 is used.

@vekterli PR
Step 1 in trying to splitt out code for testability.

### CODE REVIEW

File: vespamalloc/src/vespamalloc/malloc/datasegment.h [hunk 1]
```diff
@@ -42,7 +42,7 @@ class DataSegment
         checkAndLogBigSegment();
     }
     void enableThreadSupport() { _mutex.init(); }
-    static size_t blockId(const void * ptr)       {
+    static uint32_t blockId(const void * ptr)       {
```
Comment (vekterli) [inline vespamalloc/src/vespamalloc/malloc/datasegment.h:45]: Should this also have a return type of `BlockIdT`?

File: vespamalloc/src/vespamalloc/malloc/datasegment.h [hunk 2]
```diff
@@ -61,51 +61,59 @@ class DataSegment
...
         SizeClassT sizeClass()              const { return _sizeClass; }
-        FreeCountT realNumBlocks()          const { return _realNumBlocks; }
-        FreeCountT freeChainLength()        const { return _freeChainLength; }
+        BlockIdT realNumBlocks()          const { return _realNumBlocks; }
+        BlockIdT freeChainLength()        const { return _freeChainLength; }
         void sizeClass(SizeClassT sc)             { _sizeClass = sc; }
-        void realNumBlocks(FreeCountT fc)         { _realNumBlocks = fc; }
-        void freeChainLength(FreeCountT fc)       { _freeChainLength = fc; }
+        void realNumBlocks(BlockIdT fc)         { _realNumBlocks = fc; }
+        void freeChainLength(BlockIdT fc)       { _freeChainLength = fc; }
```
Comment (vekterli) [inline vespamalloc/src/vespamalloc/malloc/datasegment.h:79]: Nit: consider realigning these code blocks after type change

**Review by vekterli — APPROVED** [review]
:+1:

File: vespamalloc/src/vespamalloc/malloc/datasegment.h (hunk 1 above)
Comment (baldersheim) [reply vespamalloc/src/vespamalloc/malloc/datasegment.h]: Yes, fixed

File: vespamalloc/src/vespamalloc/malloc/datasegment.h (hunk 2 above)
Comment (baldersheim) [reply vespamalloc/src/vespamalloc/malloc/datasegment.h:79]: Done

**Pull request merged**